src/main.rs

```diff
@@ -0,0 +1,170 @@
+use std::{
+    fs,
+    fs::File,
+    path::PathBuf,
+    io::Read,
+    io::Error,
+    path::Path,
+    process::exit
+};
+use std::fs::DirEntry;
+use std::io::ErrorKind;
+use sha256::{try_digest};
+use flate2::{
+    read::GzDecoder,
+    Compression,
+    write::GzEncoder
+};
+use tar::Archive;
+use simple_logger::SimpleLogger;
+use log::{
+    self,
+    LevelFilter
+};
+use serde_json::{json, Value};
+use docx_rs::*;
+use clap::Parser;
+
+#[derive(Parser)]
+#[clap(name = "anonymiser")]
+struct Opt {
+    /// Input folder
+    #[clap(long, short, value_parser)]
+    input: String,
+
+    /// Output folder
+    #[clap(long, short, value_parser)]
+    output: String,
+}
+fn main() {
+    SimpleLogger::new().with_level(LevelFilter::Info).init().unwrap();
+    let opt = Opt::parse();
+    let dir_input = PathBuf::from(shellexpand::full(&opt.input).unwrap().to_string());
+    let dir_output = PathBuf::from(shellexpand::full(&opt.output).unwrap().to_string());
+    let files = list_files_in_input_dir(&dir_input).unwrap();
+    for file in files {
+        match process_package(&dir_output, &file) {
+            Ok(_) => {
+                log::info!("Processed {}", file.file_name().and_then(|name| name.to_str()).unwrap() )
+            },
+            Err(err) =>  {
+                log::error!("Error: {:?}", err);
+                exit(1);
+            }
+        };
+    }
+}
+
+fn process_package(dir_output: &PathBuf, file: &PathBuf) -> Result<(), Error> {
```

Comment:
1. I tried splitting it out but because there are so many input paths and output paths to various files, you end up with functions which take a lot of arguments and return a lot of arguments as a tuple. I couldn't find a way of doing it neatly so I've left it as is.
2. I'll leave it as file because we're not explicitly searching for tar files and I hope one day they'll stop zipping them up anyway so we can skip that step.